src/libmongoc/src/mongoc/mongoc-client-session-private.h

```diff
@@ -41,6 +41,7 @@ struct _mongoc_transaction_opt_t {
 typedef enum {
    MONGOC_SESSION_NO_OPTS = 0,
    MONGOC_SESSION_CAUSAL_CONSISTENCY = (1 << 0),
+   MONGOC_SESSION_SNAPSHOT = (2 << 0),
```

Comment:
Snapshot and causal consistency are [mutually exclusive](https://github.com/mongodb/specifications/pull/1022/files#diff-6c9d82f6b269142ff8fe9f0ef8d7c90f9e2761c947b5caf92e6820124b81fc2bR138). Given that, this enum seemed like the best place for the new snapshot sessions options.

Reply:
Ah that makes sense, so they _can_ both be set on the flag; but we'll error in that case (for now).